share/spack/qa/run-flake8-tests

```diff
@@ -36,6 +32,10 @@ if [[ ! "${changed[@]}" ]]; then
     exit 0
 fi
 
+# Move to root directory of Spack
+# Allows script to be run from anywhere
+cd "$SPACK_ROOT"
+
```

Comment:
This change fixes a bug I noticed. My method of getting `QA_DIR` provides the relative path, not the absolute path. Which is fine most of the time. But if you cd to `QA_DIR` and run it as `./run-flake8-tests`, it causes this error message:
```
$ ./run-flake8-tests 
Dependencies found.
./run-flake8-tests: line 31: ./changed_files: No such file or directory
No Python files were modified.
```
The relative directory in this case is `.`, which is useless after changing directories. The solution was to change directories _after_ running `changed_files`.